modules/iam-user/main.tf

```diff
@@ -16,6 +16,10 @@ resource "aws_iam_user_login_profile" "this" {
   pgp_key                 = var.pgp_key
   password_length         = var.password_length
   password_reset_required = var.password_reset_required
+
+  lifecycle {
+    ignore_changes = [password_reset_required]
```

Comment:
I think this fix is ok as a temporary solution while the upstream provider is not fixed properly...

Please add a comment to the [upstream provider issue](https://github.com/hashicorp/terraform-provider-aws/issues/23567) around this line so that we know why this was added in the module in the future.

Reply:
@antonbabenko Done!